### PR TITLE
Catlin: Adds Dockerfile

### PR DESCRIPTION
**Pull request by SM43**
Part of #565 

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Submitter Checklist

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

/kind misc

### CODE REVIEW

**Comment by SM43** [comment]
@vdemeester @afrittoli @sthaha  Could you please review this? :smiley_cat:

**Review by sthaha — COMMENTED** [review]
/lgtm

Will let @vdemeester approve.

File: catlin/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.14 as BUILD
+
+COPY . /build
+WORKDIR /build
+RUN GOOS=linux GARCH=amd64 CGO_ENABLED=0 \
+    go build -mod=vendor -o catlin ./cmd/catlin
+
+FROM scratch
```
Comment (vdemeester) [inline catlin/Dockerfile:8]: nit: any reason to use `scratch` ? long-term we may need to want to debug a container running this image, and/or might need to stuff like certificates, …
Any way, no need to change this now

**Review by vdemeester — COMMENTED** [review]
@SM43 any reason to vendor the dependencies ? we do have multiple go modules in the repository, and so far we didn't use `vendor` for them :sweat_smile: 
If we want to go that way (vendoring), we would need a script to make sure the `vendor` folder is up-to-date. I would rather not vendor anything, at least not while `catlin` is in this repository.

File: catlin/Dockerfile (hunk 1 above)
Comment (sthaha) [reply catlin/Dockerfile:8]: > @SM43 any reason to vendor the dependencies ? we do have multiple go modules in the repository, and so far we didn't use `vendor` for them 😅
> If we want to go that way (vendoring), we would need a script to make sure the `vendor` folder is up-to-date. I would rather not vendor anything, at least not while `catlin` is in this repository.

Lets go without vendor then :D ... I am not a big fan of vendor either but the image build is super fast 😝

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
/meow

**Pull request merged**